### ISSUE TITLE
Server aborts if a connection command becomes too big

### ISSUE THREAD

**Issue by ryvnf** [minetest/minetest#11214]
##### Minetest version

5.4.0

##### Summary

The engine will call `abort` when a sanity check fails. Among the sanity checks which are performed is a sanity check for the size of a connection command:

```cpp
sanity_check(c.data.getSize() < MAX_RELIABLE_WINDOW_SIZE*512);
```

I am not sure exactly a connection command is I assume it is basically a network packet to be transmitted to a client. The problem is that if this limit is reached the server will call `abort`. One of the cases were this condition can be reached is if the size of a players inventory exceeds this limit. Then the server will abort when trying to send the inventory data to the client.

The following error message is display when the sanity check fails.

```
2021-04-19 18:09:06: ERROR[ConnectionSend]: In thread 7f43db7fe700:
2021-04-19 18:09:06: ERROR[ConnectionSend]: minetest/src/network/connection.cpp:3f9: bool con::UDPPeer::processReliableSendCommand(con::ConnectionCommand&, unsigned int): An engine assumption 'c.data.getSize() < 0x8000*512' failed.
```

##### What should happen instead

I think the reasonable thing to do is to disconnect the client which would have received such a large packet. This is still disruptive for the client but it will not crash the server and disrupt everyone else. I also think an error message like "Disconnected. Reason: Maximum packet size exceeded" would be good.

##### Steps to reproduce

Put the following command in a mod and enable it on a devtest server. Run the `addheavyitem` command twice to reach the limit and crash the server. You also need to change inventory slots between the commands as it will replace the wield item.

```lua
minetest.register_chatcommand("addheavyitem", {
	func = function(name, param)
		local player = minetest.get_player_by_name(name)
		local item = ItemStack('basenodes:dirt')
		local s = "A"
		for i = 1, 23 do
			s = s .. s
		end
		local meta = item:get_meta()
		meta:set_string("test", s)
		minetest.get_player_by_name(name):set_wielded_item(item)
		return true, tostring(string.len(s))
	end
})
```

**Comment by sfan5** [comment]
(first reported in PR #11014)

**Comment by anon55555** [comment]
I think it would be better to drop the packet and send the player a warning chat message so accounts don't become unusable.

**Comment by ryvnf** [comment]
I think dropping the packet would probably cause other issues and make the client unusable. In the case of inventory it would mean the client does not get the inventory content updated. That might be a bit better in case they can still do `/clearinv` or something. But it is probably unlikely they will know what exactly is causing the issue.

I think kicking the player with a clear error message would be the best solution because it would be simplest to implement. This is also not something which is expected to happen with mods that are reasonable. I still think it should be handled in a better way than aborting the entire server though.

**Comment by sfan5** [comment]
Wontfix per discussion in #12280.
tl;dr a hard crash forces server admins to deal with buggy mods, other solutions have some unwanted downsides
